### PR TITLE
Added setup metadata for web config

### PR DESCRIPTION
**Pull request by docbobo**
Okay, this is not ready yet & I need some help. This PR moves the ip config into the data section and adds the appropriate metadata, so that it can be configured via the UI. There are still some pieces missing to have everything work properly with  web config, but it's a start.

Now for the problems:
- When I remove the integration via web-config, my Linux system crashes
- When I add the integration via web-config, I get some exception in the destructor of MediaPlayerUtils. 

Given that the changes are pretty minimal, I suspect that's not due to them. And I have no idea where to start looking.

### CODE REVIEW

File: roon.json.in [hunk 1]
```diff
@@ -32,5 +32,6 @@
   \"dependencies\" : [
       { \"name\" : \"app\", \"version\" : \"0.2.0\" },
       { \"name\" : \"remoteOS\", \"version\" : \"0.2.0\" }
-  ]
+  ],
```
Comment (martonborzak) [inline roon.json.in:35]: Could you please add the mdns address here? I don't know if this is correct, but I assume something similar:
`  \"mdns\": \"_roon._tcp\",`

File: setup-schema.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "additionalProperties": true,
+    "required": [
+        "ip"
+    ],
+    "properties": {
+        "ip": {
+            "$id": "#/properties/ip",
+            "type": "string",
+            "title": "IP address or hostname",
+            "description": "The IP address or hostname of your Roon Core.",
```
Comment (martonborzak) [inline setup-schema.json]: The integration checks if the port is given and appends it automatically. I think the title and description should say that port can be given as optional.

File: setup-schema.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            "default": "",
+            "examples": [
+                "192.168.1.100"
```
Comment (martonborzak) [inline setup-schema.json]: add an example for port as well:
`192.168.1.100:9100`

**Review by martonborzak — CHANGES_REQUESTED** [review]

File: roon.json.in (hunk 1 above)
Comment (docbobo) [reply roon.json.in:35]: I believe Roon  doesn't used mdns for discovery but let me  check.

File: setup-schema.json (hunk 2 above)
Comment (docbobo) [reply setup-schema.json]: I am wondering if the port should always be added. I am not aware that Roon allows you to change the port...

File: roon.json.in (hunk 1 above)
Comment (docbobo) [reply roon.json.in:35]: Nope, no mdns for Roon.

Comment (docbobo) [reply roon.json.in:35]: https://github.com/RoonLabs/node-roon-api/blob/master/sood.js

File: setup-schema.json (hunk 2 above)
Comment (docbobo) [reply setup-schema.json]: Anyway, don't want to change too much. Will add that.

File: roon.json.in (hunk 1 above)
Comment (martonborzak) [reply roon.json.in:35]: Looks like we need to implement other discovery methods :)

File: setup-schema.json (hunk 2 above)
Comment (martonborzak) [reply setup-schema.json]: I don't know much about roon, just looking at the integration code that it allows for it.

**Review by martonborzak — APPROVED** [review]

**Pull request merged**